pca.go

```diff
@@ -0,0 +1,63 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// PrincipalComponents returns the k principal component direction vectors of
+// the matrix a, projected score vectors and variance values, where variables
+// are represented in the columns and observations are the rows in the input.
+// The principal component direction vectors are found by singular value
+// decomposition of the input.
+// If no singular value decomposition is possible, pc, proj and sigma are returned
+// nil and ok is returned false. k must not be greater than the number of rows less
+// one or the number of columns in a, or PrincipalComponents will panic.
+func PrincipalComponents(a mat64.Matrix, k int) (pc, proj *mat64.Dense, vars []float64, ok bool) {
```

Comment:
You're talking about the variable sizes? That is, having something like all the columns have a variance of one? I agree that's a useful thing to have, but it feels orthogonal to the PCA call. The variable scales can be changed independently by multiplying individual variable elements by the correct number.

Note that the same is not true for data weights. Data weighting cannot be accomplished by just multiplying specific values by a number. In PCA and covariance matrix, it has to come exactly after scaling to mean zero and has to use the square root of the weights, while in other algorithms the weights come in linearly (say, mean).